File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int[] ans = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(nums[0], 0);

        for(int i=1; i<n; i++) {
            int curr = nums[i];
            int req = target - nums[i];
            if(map.containsKey(req)) {
                ans[0] = map.get(req);
                ans[1] = i;
                break;
            } else {
                map.put(nums[i], i);
            }
        }

        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> ans;
        unordered_map<int, int> map;

        for(int i=0; i<n; i++) {
            int complement = target - nums[i];
            if(map.find(complement) != map.end()) {
                ans.push_back(map[complement]);
                ans.push_back(i);
                return ans;
            }
            map[nums[i]] = i;
        }
        return ans;
    }
};